File: integration-tests/test-data/mentor.ts
import type { PrismaClient } from "~/prisma/client";
import type { DefaultArgs } from "@prisma/client/runtime/client";

export async function mentorAsync(
  tx: Omit<
    PrismaClient<never, undefined, DefaultArgs>,
    "$connect" | "$disconnect" | "$on" | "$transaction" | "$extends"
  >,
) {
  const { id } = await tx.mentor.findUniqueOrThrow({
    where: {
      email: "test_0@test.com",
    },
    select: {
      id: true,
    },
  });

  await tx.mentor.update({
    where: {
      id,
    },
    data: {
      firstName: "Luca",
      lastName: "Mara",
      mobile: "1111111",
      addressStreet: "Address street",
      addressSuburb: "Address suburb",
      addressState: "Address state",
      addressPostcode: "Address postcode",
      dateOfBirth: new Date("1986-07-22T01:44:13.938Z"),
      emergencyContactName: "Luca",
      emergencyContactNumber: "Luca",
      emergencyContactAddress: "Luca",
      emergencyContactRelationship: "Luca",
      volunteerAgreementSignedOn: new Date(),
    },
  });
}

File: Achievers.DDD/Achievers.Domain/Aggregates/Mentor.cs
namespace Achievers.Domain.Aggregates;

public sealed class Mentor : Entity, IAggregateRoot
{
    public string? AzureAdid { get; private set; }
    public string Email { get; private set; } = null!;
    public string FirstName { get; private set; } = null!;
    public string LastName { get; private set; } = null!;
    public string Mobile { get; private set; } = null!;
    public string AddressStreet { get; private set; } = null!;
    public string AddressSuburb { get; private set; } = null!;
    public string AddressState { get; private set; } = null!;
    public string AddressPostcode { get; private set; } = null!;
    public string? AdditionalEmail { get; private set; }
    public DateTime? DateOfBirth { get; private set; }
    public string? EmergencyContactName { get; private set; }
    public string? EmergencyContactNumber { get; private set; }
    public string? EmergencyContactAddress { get; private set; }
    public string? EmergencyContactRelationship { get; private set; }
    public string? NextOfKinName { get; private set; }
    public string? NextOfKinNumber { get; private set; }
    public string? NextOfKinAddress { get; private set; }
    public string? NextOfKinRelationship { get; private set; }
    public string? ProfilePicturePath { get; private set; }
    public bool? HasApprovedToPublishPhotos { get; private set; }
    public DateTime? EndDate { get; private set; }
    public DateTime? VolunteerAgreementSignedOn { get; private set; }
    public int? FrequencyInDays { get; private set; }
    public int ChapterId { get; private set; }
    public string FullName { get; private set; } = null!;
    public string? EndReason { get; private set; }
    public string? PreferredName { get; private set; }
}
